File: frontend/bff/src/application/organizations/commands/move-group.command.ts
import type { RequestContext } from '../../../cross-cutting/context/request-context.js';

/** Organograma: o grupo (com os grupos abaixo dele) passa a ficar abaixo de parentGroupId. */
export interface MoveGroupCommand {
  context: RequestContext;
  groupId: string;
  parentGroupId: string;
}

File: backend/core-service/src/Fix.Application/UseCases/Organizations/Commands/MoveGroupCommand.cs
using Fix.Application.Abstractions.Authorization;
using Fix.Application.Abstractions.Context;
using Fix.Application.Abstractions.Messaging;
using Fix.Domain.AggregateRoots.Roles;

namespace Fix.Application.Organizations.Commands;

/// <summary>Move o grupo (com os grupos abaixo dele) para baixo de outro grupo no organograma.</summary>
[RequireRole(RoleCodes.UpdateUser)]
public sealed record MoveGroupCommand(Guid UserId, Guid OrganizationId, Guid GroupId, Guid ParentGroupId)
    : ICommand, IOrganizationRequest;
